waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metrics/MonitoringConfiguration.java

```diff
@@ -15,102 +15,86 @@
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import com.codahale.metrics.JmxReporter;
-import com.codahale.metrics.MetricFilter;
-import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.ScheduledReporter;
-import com.codahale.metrics.graphite.Graphite;
-import com.codahale.metrics.graphite.GraphiteReporter;
-import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
-import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
-import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+import io.micrometer.graphite.GraphiteConfig;
+import io.micrometer.graphite.GraphiteMeterRegistry;
+import io.micrometer.graphite.GraphiteProtocol;
+import io.micrometer.jmx.JmxConfig;
+import io.micrometer.jmx.JmxMeterRegistry;
 
 import com.hotels.bdp.waggledance.conf.GraphiteConfiguration;
 
 @Configuration
 public class MonitoringConfiguration {
-  private static final Logger LOG = LoggerFactory.getLogger(MonitoringConfiguration.class);
 
-  private final Set<Closeable> reporters = new HashSet<>();
-  private @Autowired MetricRegistry metricRegistry;
-  private @Autowired GraphiteConfiguration graphiteConfiguration;
+  @Bean
+  public MeterRegistry meterRegistry(GraphiteConfiguration graphiteConfiguration) {
 
-  private <R extends Closeable> R registerReporter(R reporter) {
-    reporters.add(reporter);
-    return reporter;
-  }
+    GraphiteConfig graphiteConfig = new GraphiteConfig() {
 
-  @PreDestroy
-  public void destroy() {
-    if (reporters != null) {
-      for (Closeable reporter : reporters) {
-        try {
-          if (reporter instanceof ScheduledReporter) {
-            ((ScheduledReporter) reporter).report();
-          }
-          reporter.close();
-        } catch (Exception e) {
-          LOG.warn("Problem closing reporter", e);
-        }
+      @Override
+      public String host() {
+        return graphiteConfiguration.getHost();
       }
-    }
-  }
 
-  @PostConstruct()
-  public void init() {
-    registerBaseMetrics();
-    registerReporter(jmxReporterBuilder().build()).start();
-    if (graphiteConfiguration.isEnabled()) {
-      GraphiteReporter graphiteReporter = graphiteReporterBuilder().build(newGraphite());
-      registerReporter(graphiteReporter);
-      graphiteReporter.start(graphiteConfiguration.getPollInterval(), graphiteConfiguration.getPollIntervalTimeUnit());
-    }
-  }
+      @Override
+      public int port() {
+        return graphiteConfiguration.getPort();
+      }
 
-  private void registerBaseMetrics() {
-    metricRegistry.register("gc", new GarbageCollectorMetricSet());
```

Comment:
In the changelog we need to add what the new metrics path to each of these is (if they've changed).

Reply:
I will do that once I test it out on dev and confirm what matches with what.